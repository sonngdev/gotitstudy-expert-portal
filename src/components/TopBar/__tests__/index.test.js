import React from 'react';
import { shallow } from 'enzyme';
import { TopBar } from '../index';

describe('<TopBar />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TopBar />);
  });

  it('renders', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('contains Got It\'s logo', () => {
    const logo = wrapper.find('Logo');
    expect(logo.exists()).toBe(true);
    expect(logo.prop('name')).toBe('gotit');
  });

  it('contains expert\'s ranking', () => {
    const ranking = wrapper.find('ExpertRanking');
    expect(ranking.exists()).toBe(true);
  });

  it('contains working stats', () => {
    const stats = wrapper.find('WorkingStats');
    expect(stats.exists()).toBe(true);
  });

  it('contains current time', () => {
    const time = wrapper.find('CurrentTime');
    expect(time.exists()).toBe(true);
  });

  it('contains STOP WORKING button', () => {
    const button = wrapper.find('StopWorkingButton');
    expect(button.exists()).toBe(true);
  });

  it('contains session timer', () => {
    const timer = wrapper.find('SessionTimer');
    expect(timer.exists()).toBe(true);
  });
});
