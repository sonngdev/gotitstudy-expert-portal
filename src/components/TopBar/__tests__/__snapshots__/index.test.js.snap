// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> renders 1`] = `
"<TopMenu className=\\"u-backgroundPrimary u-paddingVerticalTiny u-paddingHorizontalMedium u-textWhite\\">
  <Logo name=\\"gotit\\" variant=\\"white\\" fileType=\\"svg\\" />
  <ExpertRanking />
  <WorkingStats />
  <SessionTimer />
  <div>
    <CurrentTime />
    <StopWorkingButton />
  </div>
</TopMenu>"
`;
